### ISSUE TITLE
Add keyboard shortcuts for moving blocks

### ISSUE THREAD

**Issue by ZebulanStanphill** [WordPress/gutenberg#10455]
## The issue
Currently, there is no way to move a block when the side controls are not shown. Ideally, there should be keyboard shortcuts for moving blocks up and down.

## Proposed solution
In #71 I noticed that the following was suggested as the macOS shortcut:

Move block up: <kbd><kbd>^Ctrl</kbd>+<kbd>⌥Option</kbd>+<kbd>↑</kbd></kbd>
Move block down: <kbd><kbd>^Ctrl</kbd>+<kbd>⌥Option</kbd>+<kbd>↓</kbd></kbd>

If these were used, the Windows and Linux equivalents would be:

Move block up: <kbd><kbd>⇧Shift</kbd>+<kbd>⎇Alt</kbd>+<kbd>↑</kbd></kbd>
Move block down: <kbd><kbd>⇧Shift</kbd>+<kbd>⎇Alt</kbd>+<kbd>↓</kbd></kbd>

**Comment by mtias** [comment]
Ulysses uses `⌃⌘↑` and `⌃⌘↓` for this functionality.

**Comment by ZebulanStanphill** [comment]
@mtias What would the Linux/Windows equivalent of those be?

**Comment by afercia** [comment]
Related #11154 and #3218.

**Comment by ervind** [comment]
Are there any moves forward with this one?

It would really mean improvements in usability. Imagine a 2000 word-long article where you have to duplicate some paragraphs and move them to the bottom.

Now I have 2 options:
1) drag&drop
2) clicking on the down arrow a million times

It would be a lot more comfortable if I could move it up/down with pressing some keys on my keyboard (e.g. simulating the clicks on the arrows).

**Comment by daveredfern** [comment]
+1 Amazed this isn’t implemented yet.

**Comment by ckaden** [comment]
I frequently stumble upon this issue as well. The main issue is when the arrow icons change their position on the display slightly, leading to a lot of missclicks.

**Comment by zieliz** [comment]
I would love this! I don't want to keep clicking.

**Comment by jeremyfelt** [comment]
> Ulysses uses ⌃⌘↑ and ⌃⌘↓ for this functionality.Dropbox Paper also uses Ulysses uses ⌃⌘↑ and ⌃⌘↓ for this functionality.

**Comment by ellatrix** [comment]
> Now I have 2 options:
> 
> 1. drag&drop
> 2. clicking on the down arrow a million times

There's also copy paste.

**Comment by ZebulanStanphill** [comment]
It's worth noting that, on a keyboard, after you've focused one of the movers, you can just press <kbd>Space</kbd> to press it. (This works for all HTML `<button>`s, actually.)

**Comment by mtias** [comment]
> There's also copy paste.

And this one is in progress: https://github.com/WordPress/gutenberg/issues/22031
